### PR TITLE
Display events list widget on frontend

### PR DESCRIPTION
**Pull request by paulmskim**
**Ticket:** [TEC-3595]

This PR sets up the FE to allow for widgets to display with dummy data. There is nothing hooked up yet, just the FE template.

**Artifact:**

image

[TEC-3595]: https://moderntribe.atlassian.net/browse/TEC-3595

### CODE REVIEW

File: src/views/v2/widgets/events-list.php [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+ * @var string   $rest_nonce         The REST nonce.
+ * @var int      $should_manage_url  int containing if it should manage the URL.
+ * @var string[] $container_classes  Classes used for the container of the view.
+ * @var array    $container_data     An additional set of container `data` attributes.
+ * @var string   $breakpoint_pointer String we use as pointer to the current view we are setting up with breakpoints.
+ * @var array    $messages           An array of user-facing messages, managed by the View.
+ */
+
+$events = [
+
```
Comment (tr1b0t) [inline src/views/v2/widgets/events-list.php:25]: * :no_entry_sign: Empty array declaration must have no space between the parentheses
* :no_entry_sign: Empty array declaration must have no space between the parentheses

<sub>hash: 3b761c1ddc756a1c46cde6fca7b7290e</sub>

File: src/views/v2/widgets/events-list.php [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+];
+$rest_url = 'https://tri.be/';
+$rest_nonce = 'a3ghv98awe98';
```
Comment (tr1b0t) [inline src/views/v2/widgets/events-list.php:28]: :warning: Equals sign not aligned with surrounding assignments; expected 11 spaces but found 1 space

<sub>hash: 5e6ecdfd3cbbcfd0b79d15e5ee03fa87</sub>

File: src/views/v2/widgets/events-list.php [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+$should_manage_url = '0';
```
Comment (tr1b0t) [inline src/views/v2/widgets/events-list.php:29]: :warning: Equals sign not aligned with surrounding assignments; expected 9 spaces but found 1 space

<sub>hash: 6e97355dfa563c8f3e39696eeba8b624</sub>

File: src/views/v2/widgets/events-list.php [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+$container_classes = [ 'tribe-common', 'tribe-events', 'tribe-events-view', 'tribe-events-widget', 'tribe-events-widget--events-list' ];
```
Comment (tr1b0t) [inline src/views/v2/widgets/events-list.php:30]: :warning: Equals sign not aligned with surrounding assignments; expected 2 spaces but found 1 space

<sub>hash: 86ecf8ce1ce9d969afd5a93d81cfac9d</sub>

File: src/views/v2/widgets/events-list.php [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+$container_data = [];
```
Comment (tr1b0t) [inline src/views/v2/widgets/events-list.php:31]: :warning: Equals sign not aligned with surrounding assignments; expected 2 spaces but found 1 space

<sub>hash: bf0d82455ab960d85745b89700971589</sub>

File: src/views/v2/widgets/events-list.php [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+$breakpoint_pointer = 'e1a5c9c2-2781-4fcb-b4fa-0ab738292e04';
```
Comment (tr1b0t) [inline src/views/v2/widgets/events-list.php:32]: :warning: Equals sign not aligned with surrounding assignments; expected 5 spaces but found 1 space

<sub>hash: 116a9bf1e39171a9fae32a29d0a97ce2</sub>

File: src/views/v2/widgets/events-list.php [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+$messages = [];
+?>
```
Comment (tr1b0t) [inline src/views/v2/widgets/events-list.php:34]: :warning: Equals sign not aligned with surrounding assignments; expected 11 spaces but found 1 space

<sub>hash: 19b0f46e4d46de332acb819f4f4e0534</sub>

File: src/views/v2/widgets/events-list.php [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+<div
+	<?php tribe_classes( $container_classes ); ?>
+	data-js="tribe-events-view"
+	data-view-rest-nonce="<?php echo esc_attr( $rest_nonce ); ?>"
+	data-view-rest-url="<?php echo esc_url( $rest_url ); ?>"
+	data-view-manage-url="<?php echo esc_attr( $should_manage_url ); ?>"
+	<?php foreach ( $container_data as $key => $value ) : ?>
```
Comment (tr1b0t) [inline src/views/v2/widgets/events-list.php:41]: :no_entry_sign: Wrong escaping function. href and src attributes should be escaped by `esc_url()`, not by `esc_attr()`.

<sub>hash: c255bca888fa1cd539994f29449aa063</sub>

File: src/views/v2/widgets/events-list.php [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+		data-view-<?php echo esc_attr( $key ) ?>="<?php echo esc_attr( $value ) ?>"
+	<?php endforeach; ?>
```
Comment (tr1b0t) [inline src/views/v2/widgets/events-list.php:43]: * :no_entry_sign: Inline PHP statement must end with a semicolon
* :no_entry_sign: Inline PHP statement must end with a semicolon

<sub>hash: 72a344540ebae214f5f14bb377d4f91f</sub>

File: src/views/v2/widgets/events-list.php [hunk 10]
```diff
@@ -0,0 +1,61 @@
...
+	<?php if ( ! empty( $breakpoint_pointer ) ) : ?>
+		data-view-breakpoint-pointer="<?php echo esc_attr( $breakpoint_pointer ); ?>"
+	<?php endif; ?>
+>
+	<div class="tribe-events-widget-events-list">
+		<header class="tribe-events-widget-events-list__header">
+			<h3 class="tribe-events-widget-events-list__header-title"><?php // title ?></h3>
+		</header>
```
Comment (tr1b0t) [inline src/views/v2/widgets/events-list.php:51]: :no_entry_sign: Inline comments must end in full-stops, exclamation marks, or question marks

<sub>hash: 0dced780f1a473afed6ef776944947e0</sub>

File: src/views/v2/widgets/events-list.php [hunk 11]
```diff
@@ -0,0 +1,61 @@
...
+
+		<?php if ( ! empty( $events ) ) : ?>
+			<?php foreach ( $events as $event ) : ?>
+				<?php // include event template ?>
+			<?php endforeach; ?>
```
Comment (tr1b0t) [inline src/views/v2/widgets/events-list.php:56]: :no_entry_sign: Inline comments must end in full-stops, exclamation marks, or question marks

<sub>hash: 59453438f74f22d262897de50d939335</sub>

File: src/views/v2/widgets/events-list.php [hunk 12]
```diff
@@ -0,0 +1,61 @@
...
+		<?php else : ?>
+			<?php // get messages component ?>
+		<?php endif; ?>
```
Comment (tr1b0t) [inline src/views/v2/widgets/events-list.php:59]: :no_entry_sign: Inline comments must end in full-stops, exclamation marks, or question marks

<sub>hash: ebd5a029570e13ffe76cf56e76aaa47f</sub>

**Review by tr1b0t — COMMENTED** [review]
:no_entry_sign: 8 error(s)
:warning: 6 warning(s)

File: src/views/v2/widgets/events-list.php (hunk 1 above)
Comment (paulmskim) [reply src/views/v2/widgets/events-list.php:25]: this is fixed in a later PR.

File: src/views/v2/widgets/events-list.php (hunk 8 above)
Comment (paulmskim) [reply src/views/v2/widgets/events-list.php:41]: this is not an issue.

File: src/views/v2/widgets/events-list.php (hunk 10 above)
Comment (paulmskim) [reply src/views/v2/widgets/events-list.php:51]: this is removed in a later PR.

File: src/views/v2/widgets/events-list.php (hunk 11 above)
Comment (paulmskim) [reply src/views/v2/widgets/events-list.php:56]: this is removed in a later PR.

File: src/views/v2/widgets/events-list.php (hunk 12 above)
Comment (paulmskim) [reply src/views/v2/widgets/events-list.php:59]: this is removed in a later PR.

File: src/views/v2/widgets/events-list.php [hunk 13]
```diff
@@ -0,0 +1,61 @@
...
+ * Override this template in your own theme by creating a file at:
+ * [your-theme]/tribe/events/v2/widgets/events-list.php
+ *
+ * See more documentation about our views templating system.
+ *
+ * @link http://m.tri.be/1aiy
+ *
+ * @version TBD
+ *
+ * @var array    $events             The array containing the events.
```
Comment (lucatume) [inline src/views/v2/widgets/events-list.php:14]:
```suggestion
 * @var array<\WP_Post>    $events             The array containing the events.
```

File: src/views/v2/widgets/events-list.php [hunk 14]
```diff
@@ -0,0 +1,61 @@
...
+ * @var string   $rest_url           The REST URL.
+ * @var string   $rest_nonce         The REST nonce.
+ * @var int      $should_manage_url  int containing if it should manage the URL.
+ * @var string[] $container_classes  Classes used for the container of the view.
```
Comment (lucatume) [inline src/views/v2/widgets/events-list.php:18]:
```suggestion
 * @var array<string> $container_classes  Classes used for the container of the view.
```

File: src/views/v2/widgets/events-list.php [hunk 15]
```diff
@@ -0,0 +1,61 @@
...
+ * @var array    $container_data     An additional set of container `data` attributes.
```
Comment (lucatume) [inline src/views/v2/widgets/events-list.php:19]:
```suggestion
 * @var array<string,mixed>    $container_data     An additional set of container `data` attributes.
```

File: src/views/v2/widgets/events-list.php [hunk 16]
```diff
@@ -0,0 +1,61 @@
...
+ * @var string   $breakpoint_pointer String we use as pointer to the current view we are setting up with breakpoints.
+ * @var array    $messages           An array of user-facing messages, managed by the View.
```
Comment (lucatume) [inline src/views/v2/widgets/events-list.php:21]:
```suggestion
 * @var array<string,string>    $messages           An array of user-facing messages, managed by the View.
```

**Review by lucatume — CHANGES_REQUESTED** [review]

**Comment by paulmskim** [comment]
@lucatume suggestions addressed, tests should pass, this is good for another review.

**Pull request merged**